src/features/import/hooks/useImportedFile.ts

```diff
@@ -0,0 +1,42 @@
+import { addFiles } from '../store';
+import { parseExcelFile } from '../utils/parseFile';
+import { useAppDispatch } from 'core/hooks';
+import { useState } from 'react';
+import { ImportedFile, parseCSVFile } from '../utils/parseFile';
+
+export default function useFileImport() {
+  const [loading, setLoading] = useState(false);
+  const dispatch = useAppDispatch();
+
+  function parseData(file: File) {
```

Comment:
I would make this `async` and use `await` instead of `.then()` in the function, which makes it easier to read.